server/src/main/java/io/crate/analyze/relations/FullQualifiedNameFieldProvider.java

```diff
@@ -101,7 +102,12 @@ public Symbol resolveField(QualifiedName qualifiedName, @Nullable List<String> p
             tableNameMatched = true;
 
             AnalyzedRelation relation = entry.getValue();
-            Symbol newField = relation.getField(columnIdent, operation, errorOnUnknownObjectKey);
+            Symbol newField = null;
+            try {
+                newField = relation.getField(columnIdent, operation, errorOnUnknownObjectKey);
+            } catch (ColumnUnknownException e) {
+                lastColUnknownException = e;
```

Comment:
Thanks for reviewing. I can see that my current implementation will always favour `ColumnUnknownException` from `TableFunctionRelation` regardless of the order of `sources`.
I have made the change from `lastColumnUnknownException` to `firstColumnUnknownException`, but it will depend on the order of `sources` not left to right order as provided in sql since `sources` is a `map`.

Reply:
Yep, I see, somehow I thought we also use it in the user facing message, but it's only used for the privilege check, thank you!